src/components/MessageList/MessageList.tsx

```diff
@@ -556,60 +565,67 @@ const MessageListWithContext = <
-            {insertInlineUnreadIndicator && <InlineUnreadIndicator />}
-          </>
-        );
-      }
+    const wrapMessageInTheme =
+      client.userID === message.user?.id && !!myMessageTheme;
+    if (wrapMessageInTheme) {
       return (
         <>
-          <Message
-            goToMessage={goToMessage}
-            groupStyles={
-              isMessagesWithStylesAndReadBy(message) ? message.groupStyles : []
-            }
-            lastReceivedId={
-              lastReceivedId === message.id ? lastReceivedId : undefined
-            }
-            message={message}
-            onThreadSelect={onThreadSelect}
-            showUnreadUnderlay={showUnreadUnderlay}
-            style={styles.messagePadding}
-            targetedMessage={targetedMessage === message.id}
-            threadList={threadList}
-          />
+          <ThemeProvider mergedStyle={modifiedTheme}>
+            <Message
+              goToMessage={goToMessage}
+              groupStyles={
+                isMessageWithStylesReadByAndDateSeparator(message)
+                  ? message.groupStyles
+                  : []
+              }
+              lastReceivedId={
+                lastReceivedId === message.id ? lastReceivedId : undefined
+              }
+              message={message}
+              onThreadSelect={onThreadSelect}
+              showUnreadUnderlay={showUnreadUnderlay}
+              style={styles.messagePadding}
+              targetedMessage={targetedMessage === message.id}
+              threadList={threadList}
+            />
+          </ThemeProvider>
+          {isMessageWithStylesReadByAndDateSeparator(message) &&
+            message.dateSeparator && (
+              <InlineDateSeparator date={message.dateSeparator} />
+            )}
           {/* Adding indicator below the messages, since the list is inverted */}
           {insertInlineUnreadIndicator && <InlineUnreadIndicator />}
         </>
       );
     }
 
-    return null;
+    return (
+      <>
+        <Message
+          goToMessage={goToMessage}
+          groupStyles={
+            isMessageWithStylesReadByAndDateSeparator(message)
+              ? message.groupStyles
+              : []
+          }
+          lastReceivedId={
+            lastReceivedId === message.id ? lastReceivedId : undefined
+          }
+          message={message}
+          onThreadSelect={onThreadSelect}
+          showUnreadUnderlay={showUnreadUnderlay}
+          style={styles.messagePadding}
+          targetedMessage={targetedMessage === message.id}
+          threadList={threadList}
+        />
+        {isMessageWithStylesReadByAndDateSeparator(message) &&
+          message.dateSeparator && (
+            <InlineDateSeparator date={message.dateSeparator} />
+          )}
+        {/* Adding indicator below the messages, since the list is inverted */}
+        {insertInlineUnreadIndicator && <InlineUnreadIndicator />}
```

Comment:
not a huge deal, more stylistic consistency comment.

the `if (wrapMessageTheme) {` body has this same return so do we either want to create a const for it or remove the `if` and have a `return` with a ternary for the first part of both trees or leave it in this style?

Reply:
changed to ternary